src/lib.rs

```diff
@@ -164,6 +254,34 @@ fn check_if_input_text_is_plaintext(text: &str) -> CheckResult {
     }
 }
 
+/// Stores a successful DecoderResult into the cache table
+fn success_result_to_cache(
+    text: &String,
+    start_time: SystemTime,
+    result: &DecoderResult,
+) -> Result<usize, rusqlite::Error> {
+    let stop_time = SystemTime::now();
+    let execution_time_ms: i64 = match stop_time.duration_since(start_time) {
+        Ok(duration) => duration.as_millis().try_into().unwrap_or(-2),
+        Err(_) => {
+            cli_pretty_printing::warning(
+                "Stop time is less than start time. Clock may have gone backwards.",
+            );
+            -1
+        }
+    };
+    let cache_entry = storage::database::CacheEntry {
+        encoded_text: String::from(text),
+        decoded_text: match result.text.last() {
+            Some(d_text) => String::from(d_text),
+            None => String::new(),
+        },
+        path: result.path.clone(),
+        execution_time_ms,
```

Comment:
did i tell u to add this or were u thinking of execution time per decoder 🤔 

this is still cute, we could potentially tell how much time we saved the user

Reply:
I think you did ask me, but also it's in the ``database_implementation.md`` doc

Reply:
hahahaha amazing ty